pipeline.py

```diff
@@ -72,32 +87,110 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        for article in self.corpus_manager.get_articles().values():
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+
+            tokens_for_article = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+
+            for token in tokens:
+                tokens_for_article.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+                multiple_tagged_tokens.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(tokens_for_article), 'cleaned')
+            article.save_as(' '.join(single_tagged_tokens), 'single_tagged')
+            article.save_as(' '.join(multiple_tagged_tokens), 'multiple_tagged')
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        cleaned_text = ' '.join(re.findall(r'[а-яёА-ЯЁ]+', raw_text))
+        analyzed_text = Mystem().analyze(cleaned_text)
+        morph = pymorphy2.MorphAnalyzer()
+
+        tokens = []
+        for token in analyzed_text:
+
+            if 'analysis' not in token:
+                continue
+            if not token['analysis']:
+                continue
+
+            morph_token = MorphologicalToken(token['text'])
+            morph_token.normalized_form = token['analysis'][0].get('lex')
+            morph_token.tags_mystem = token['analysis'][0].get('gr')
+            tokens.append(morph_token)
+            morph_token.tags_pymorphy = morph.parse(token['text'])[0].tag
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+    if not path.exists():
+        raise FileNotFoundError
+
+    if not path.is_dir():
+        raise NotADirectoryError
+
+    all_article_ids = []
+
+    for file in path.glob("*.txt"):
+        with open(file, 'r', encoding='utf-8') as text_file:
+            text = text_file.read()
+        raw_data = list(path.glob('*_raw.txt'))
+        meta_data = list(path.glob('*_meta.json'))
+
+        if not len(meta_data) == len(raw_data):
+            raise InconsistentDatasetError
```

Comment:
could you please explain why you have to verify this condition within a cycle over files

Reply:
To make sure that i got raw texts from all of the files that i have